packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/InAppPurchasePluginTests.m

```diff
@@ -105,22 +105,17 @@ - (void)testPaymentQueueStorefrontReturnsNil {
 }
 
 - (void)testGetProductResponse {
-  XCTestExpectation *expectation =
-      [self expectationWithDescription:@"expect response contains 1 item"];
-  FlutterMethodCall *call = [FlutterMethodCall
-      methodCallWithMethodName:@"-[InAppPurchasePlugin startProductRequest:result:]"
-                     arguments:@[ @"123" ]];
-  __block id result;
-  [self.plugin handleMethodCall:call
-                         result:^(id r) {
-                           [expectation fulfill];
-                           result = r;
-                         }];
-  [self waitForExpectations:@[ expectation ] timeout:5];
-  XCTAssert([result isKindOfClass:[NSDictionary class]]);
-  NSArray *resultArray = [result objectForKey:@"products"];
-  XCTAssertEqual(resultArray.count, 1);
-  XCTAssertTrue([resultArray.firstObject[@"productIdentifier"] isEqualToString:@"123"]);
+  NSArray *argument = @[ @"123" ];
+  [self.plugin
+      startProductRequestProductIdentifiers:argument
+                                 completion:^(SKProductsResponseMessage *_Nullable response,
```

Comment:
Can you add an assertion that the completion block was called too?  Here's an example of that: https://github.com/flutter/packages/pull/6069/files#diff-3033f32996418c96cac1b73875958073e57facf79ffdbc7e6864f8b27a63de37R21